crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -793,6 +793,39 @@ impl Starknet {
         }
     }
 
+    pub fn update_gas(&mut self, gas_prices: GasUpdate) -> DevnetResult<GasUpdate> {
```

Comment:
return type of the method can be only GasUpdate.

Reply:
I don't get it. What is the problem with that? Can you explain it further?

Reply:
There is no error propagation in the method implementation. Each of the function calls, conversions is returning a type not a Result<type>. This is the reason why there is no need for the return type to be DevnetResult. None of the code paths can return an Error result

Reply:
Ok, but now it has changed with generate block functionality.

Reply:
at the time of writing my comments there were no such functionality

Reply:
yes I know :)